src/ngx_http_lua_ssl_module.c

```diff
@@ -303,9 +322,15 @@ ngx_http_lua_ffi_update_last_ticket_decryption_key(SSL_CTX *ctx,
     pkey = &pkey[keys->nelts-1];
 
     dd("replace the last key");
-    ngx_memcpy(pkey->name, key, 16);
-    ngx_memcpy(pkey->aes_key, key + 16, 16);
-    ngx_memcpy(pkey->hmac_key, key + 32, 16);
+    if (ngx_strlen(key) == 48){
```

Comment:
@fbomlisboa It should not assume a `\0` at the end of the string in the first place. The caller should pass down the size explicitly. In this case, the caller is the Lua code and Lua strings are not C strings per se and use of the `#` operator on Lua strings is an `O(1)` operation already. The Lua code should pass the result of `#str` as a separate argument to this C function.